Tools/PostProcessing/plot_timestep_duration.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import re
+import sys
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def extract_data(filename):
+
+    regex_step = re.compile(
```

Comment:
Using `re` is good here. Just to comment, when I do this kind of thing, I use more basic string manipulation (which I find easier to write than composing an `re` string), like:
"""
line = f.read()
if line.startswith('Evolve time'):
    sline = line.split(' ')
    time_data.append(float(sline[8])
"""

Reply:
Hm, I think the regrex here is more expressive than this (e.g., shows exact structure of what we expect, no magic number, multiple lines, etc.) :)
I would keep the regex.